parla/tasks.py

```diff
@@ -418,7 +418,7 @@ def decorator(body):
         taskid.dependencies = dependencies
 
         # gather input/output/inout, which is hint for data from or to the this task
-        dataflow = Dataflow(input, output, inout)
+        dataflow = Dataflow(list(input), list(output), list(inout))
```

Comment:
Why was this added? 

Also side comment for the future, I think we should add a flatten for input/output/inout in `spawn`. Just in case the user passes a list of lists, etc. (2-matrix of parrays)

Reply:
disregard second part. Its fine if this is on the user, its too expensive to put in every spawn calll.

Reply:
I wanted to iterate through `input + inout` or `output + inout`, etc., in various places, but tuples can't be concatenated, so I just moved them to lists.

I suppose that could cause a perf hit... Do you think I should just go rework all of those into separate `for` loops?

Reply:
Maybe refactor to do the concatenation once at dataflow __init__ ?

Reply:
I don't think its the major source of overhead right now, we've got a death by a thousand slightly larger cuts first.

Reply:
Added as a TODO comment for later